playbook.yml

```
@@ -56,6 +56,33 @@
         - "{{ dns_replace_address }}"
       loop_control:
         label: "{{ item.1.name }}: {{ item.1.url }}"
+    
+    - name: Remove unwanted lines containing ip6- 
+      ansible.builtin.lineinfile:
+        path: "lists/{{ item.0 }}/{{ item.1.type }}/{{ item.1.name }}"
+        regexp: ' ip6-'
+        state: absent
+      loop: "{{ group_names | product(dns_adblock_lists) | list }}"
+      loop_control:
+        label: "{{ item.1.name }}: {{ item.1.url }}"
+    
+    - name: Remove custom hosts IP addresses in files 
+      ansible.builtin.lineinfile:
+        path: "lists/{{ item.0 }}/{{ item.1.type }}/{{ item.1.name }}"
+        regexp: '([0-9]{1,3}\.){3}[0-9]{1,3}'
+        state: absent
+      loop: "{{ group_names | product(dns_adblock_lists) | list }}"
+      loop_control:
+        label: "{{ item.1.name }}: {{ item.1.url }}"
+
+    - name: Remove custom hosts localhost in files
+      ansible.builtin.lineinfile:
+        path: "lists/{{ item.0 }}/{{ item.1.type }}/{{ item.1.name }}"
+        regexp: ' localhost'
+        state: absent
+      loop: "{{ group_names | product(dns_adblock_lists) | list }}"
+      loop_control:
+        label: "{{ item.1.name }}: {{ item.1.url }}"
```

```suggestion
    
    - name: Remove unwanted lines
      ansible.builtin.lineinfile:
        path: "lists/{{ item.0 }}/{{ item.1.type }}/{{ item.1.name }}"
        regexp: "{{ item.2 }}"
        state: absent
      with_nested:
        - "{{ group_names }}"
        - "{{ dns_adblock_lists }}"
        - "{{ _regexp }}"
      vars:
        _regexp:
          - ' ip6-'
          - '([0-9]{1,3}\.){3}[0-9]{1,3}'
          - ' localhost'
          - '#'
      loop_control:
        label: "{{ item.1.name }}: {{ item.1.url }}"
```

We _could_ replace these tasks with this shorter version, but it would not make it any faster. So let's leave it for now!

---

Cool :)